src/Service/Manager/SponsorshipBenefitManager.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\Manager;
+
+use App\Entity\SponsorshipBenefit;
+use App\Repository\SponsorshipBenefitRepository;
+use App\Service\Assembler\SponsorshipBenefitAssembler;
+use App\Service\DTO\SponsorshipBenefitData;
+use Doctrine\ORM\EntityManagerInterface;
+
+/**
+ * @method SponsorshipBenefitRepository  getRepository()
+ */
+final class SponsorshipBenefitManager extends EntityManager implements SponsorshipBenefitManagerInterface
+{
+    public function __construct(EntityManagerInterface $em, SponsorshipBenefitAssembler $assembler)
+    {
+        parent::__construct($em, $assembler, SponsorshipBenefit::class);
+    }
+
+    public function switchPosition(string $move, string $id): SponsorshipBenefitData
+    {
+        $sponsorshipBenefit = $this->getRepository()->find($id);
+
+        $first_position = $sponsorshipBenefit->getPosition();
+        $position = $move == 'up' ? $sponsorshipBenefit->getPosition() - 1 : $sponsorshipBenefit->getPosition() + 1;
```

Comment:
you should better use pre-increment and pre-decrement

Reply:
can't do this here because i need to set :

Old level with $first_position
New level with $position
So if i use pre-increment / pre-decrement, my first position is not valid when i want to use it